packages/voice/src/models/v1/helper.ts

```diff
@@ -0,0 +1,239 @@
+
+export class IceSvamletBuilder extends SvamletBuilder<IceSvamlAction, IceSvamlInstruction, IceResponse> {}
+
+export class PieSvamletBuilder extends SvamletBuilder<PieSvamlAction, PieSvamlInstruction, PieResponse> {}
+
+export const customCalloutHelper = {
+  formatIceResponse: (action?: IceSvamlAction, ...instructions: IceSvamlInstruction[]): string => {
+    const iceResponse: IceResponse = {
+      action,
+      instructions: instructions.length > 0 ? instructions : undefined,
+    };
+    return JSON.stringify(iceResponse);
+  },
+  formatAceResponse:(action?: AceSvamlAction, ...instructions: AceSvamlInstruction[]): string => {
+    const aceResponse: AceResponse = {
+      action,
+      instructions: instructions.length > 0 ? instructions : undefined,
+    };
+    return JSON.stringify(aceResponse);
+  },
+  formatPieResponse:(action?: PieSvamlAction, ...instructions: PieSvamlInstruction[]): string => {
+    const pieResponse: PieResponse = {
+      action,
+      instructions: instructions.length > 0 ? instructions : undefined,
+    };
+    return JSON.stringify(pieResponse);
+  },
+};
+
+export const svamlActionHelper = {
+  buildConnectConf: (connectConfProps: ConnectConfProps): SvamlActionConnectConf => {
+    return {
+      name: 'connectConf',
+      ...connectConfProps,
+    };
+  },
+  buildConnectMxp: (connectMxpProps: ConnectMxpProps): SvamlActionConnectMxp => {
+    return {
+      name: 'connectMxp',
+      ...connectMxpProps,
+    };
+  },
+  buildConnectPstn: (connectPstnProps: ConnectPstnProps): SvamlActionConnectPstn => {
+    return {
+      name: 'connectPstn',
+      ...connectPstnProps,
+    };
+  },
+  buildConnectSip: (connectSipProps: ConnectSipProps): SvamlActionConnectSip => {
+    return {
+      name: 'connectSip',
+      ...connectSipProps,
+    };
+  },
+  buildContinue: (): SvamlActionContinue => {
+    return {
+      name: 'continue',
+    };
+  },
+  buildHangup: (): SvamlActionHangup => {
+    return {
+      name: 'hangup',
+    };
+  },
+  buildPark: (parkProps: ParkProps): SvamlActionPark => {
+    return {
+      name: 'park',
+      ...parkProps,
+    };
+  },
+  buildRunMenu: (runMenuProps: RunMenuProps): SvamlActionRunMenu => {
+    return {
+      name: 'runMenu',
+      ...runMenuProps,
+    };
+  },
+};
+
+export const svamlInstructionHelper = {
+  buildAnswer: (): SvamlInstructionAnswer => {
+    return {
+      name: 'answer',
+    };
+  },
+  buildPlayFiles: (ids: string[], locale: TtsVoice): SvamlInstructionPlayFiles => {
+    // TODO - check for server limitations
+    return {
+      name: 'playFiles',
+      ids,
+      locale,
+    };
+  },
+  buildSay: (text: string, locale?: TtsVoice): SvamlInstructionSay => {
+    return {
+      name: 'say',
+      text,
+      locale,
+    };
+  },
+  buildSendDtmf: (dtmfValue: string): SvamlInstructionSendDtmf => {
+    const dtmfRegex = /^[0-9#w]+$/;
+    if(!dtmfRegex.test(dtmfValue)) {
```

Comment:
Java SDK is having a relax mode regarding validation and up to user to activate it (strict mode is the default one)
So still in place with log but won't throw exception
Idea is, in case of user do not have possibility to fix the DTMF source value, to avoid a blocking situation at user level because of SDK just refuse to perform the request
Java SDK implementation: https://github.com/sinch/sinch-sdk-java/blob/main/client/src/main/com/sinch/sdk/models/DualToneMultiFrequency.java

Reply:
I'm wondering if we should block the DTMF action at SDK level (and not perform the intended action), or let the action to occur on the backend (and fail), which would have the exact same experience to the end user on the line.
Logging the error is the most important thing to do to help the developer understand what may have gone wrong.
BTW, there is a typo in your error message on line 35: "Invalid DTFM format" (meta review!!😁)

Reply:
As far as I know, currently, server could silently ignore inconsistent value
(thank you for the Java review 😄 )